tests/test_functional.py

```diff
@@ -296,9 +296,72 @@ def test_logout_no_confirmation(client):
     assertRedirects(response, reverse("accounts:login"))
 
 
+def test_logout_no_confirmation_when_already_session_and_tokens_already_expired(client):
+    """Logout without confirmation requires the id_token"""
+
+    user = UserFactory()
+    ApplicationFactory(client_id=OIDC_PARAMS["client_id"])
+
+    with freeze_time("2023-05-25 9:34"):
+        auth_url = reverse("oidc_overrides:authorize")
+        auth_complete_url = add_url_params(auth_url, OIDC_PARAMS)
+        response = client.get(auth_complete_url)
+        assertRedirects(response, reverse("accounts:login"))
+
+        response = client.post(response.url, data={"email": user.email, "password": DEFAULT_PASSWORD})
+        assert get_user(client).is_authenticated is True
+        response = client.get(response.url)
+        auth_response_params = get_url_params(response.url)
+        id_token = oidc_flow_followup(client, auth_response_params, user)
+
+    with freeze_time("2023-05-25 20:05"):
+        assert get_user(client).is_authenticated is False
+        response = call_logout(
+            client, "get", {"id_token_hint": id_token, "post_logout_redirect_uri": "https://callback/"}
+        )
+
+        assertRedirects(response, "https://callback/", fetch_redirect_response=False)
+        assert not get_user(client).is_authenticated
+        assert token_are_revoked(user)
+
+    response = client.get(auth_complete_url)
+    assertRedirects(response, reverse("accounts:login"))
```

Comment:
Autant les laisser dans le block avec le temps figé, autrement ça utilise le temps courant, des jours/mois/années après le temps du test.
```suggestion
        response = client.get(auth_complete_url)
        assertRedirects(response, reverse("accounts:login"))
```